sniffles/rulereader.py

```diff
@@ -654,27 +706,61 @@ def __init__(self, protocol=None):
             self.sport = self.background_traffic
             self.dport = 'any'
             if self.background_traffic == 'http':
-                self.contentString += 'HTTP/1.1 200 OK\r\n'
+                self.response = random.choice(self.httpResponseCodes)
+                self.contentString += 'HTTP/1.1 ' + self.response + \
+                                      '\r\n\r\n'
             elif self.background_traffic == 'ftp':
-                self.contentString += '220 FTP server ready\r\n'
+                self.response = random.choice(self.ftpResponseCodes)
+                self.contentString += ' '.join([self.response, cr_lf])
             elif self.background_traffic == 'pop':
-                self.contentString += '+OK \r\n'
+                self.response = random.choice(self.popResponseCodes)
+                self.contentString += ' '.join([self.response, cr_lf])
             elif self.background_traffic == 'mail':
-                self.contentString += '220 \r\n'
+                # Further select ports for accurate contents
+                self.sport = str(random.choice(MAIL_PORTS))
+                # SMTP
+                if self.sport == '25' or self.sport == '465':
+                    self.response = random.choice(self.smtpResponseCodes)
+                    self.contentString += ' '.join([self.response, 
+                                                    cr_lf])
+                # IMAP
+                elif self.sport == '143':
+                    self.response = random.choice(self.imapResponseCodes)
+                    self.contentString += ' '.join([imapTag, 
+                                         self.response, cr_lf])
         # Set a rule to send ACK when client -> server
         # only in http, content is generated
         elif self.flow == 'to server':
             self.sport = 'any'
             self.dport = self.background_traffic
             if self.background_traffic == 'http':
-                self.contentString += 'GET / HTTP/1.1\r\n\r\n'
-                self.contentString += 'Host: \r\n'
-                self.contentString += 'content-type: text-html\r\n'
-                self.contentString += 'text\r\n'
+                self.request = random.choice(self.httpRequestCodes)
+                self.url = random.choice(self.httpURL)
+                self.contentString += self.request
+                self.contentString += ' / HTTP/1.1\r\n'
+                self.contentString += 'Host: ' + self.url + '\r\n\r\n'
+            elif self.background_traffic == 'ftp':
+                self.request = random.choice(self.ftpRequestCodes)
+                self.contentString += ' '.join([self.request, cr_lf])
+            elif self.background_traffic == 'pop':
+                self.request = random.choice(self.popRequestCodes)
+                self.contentString += ' '.join([self.request, cr_lf])
+            elif self.background_traffic == 'mail':
+                self.dport = str(random.choice(MAIL_PORTS))
```

Comment:
Better to separate IMAP from SMTP, since we already have a separate entry for POP in this `if` statement. SMTP is used to *send* emails, while IMAP and POP are used by an email client to download emails from the server.